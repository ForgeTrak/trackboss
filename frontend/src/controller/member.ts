import generateHeaders from "./utils";


export async function createMember(token: string, memberData: object) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/new`, {
        method: 'POST',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(memberData),
    });
    return response.json();
}

export async function getMember(token: string, memberId: number) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/${memberId}`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function getMemberList(token: string, listType?: string) {
    if (listType) {
        const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/list?status=${listType}`, {
            method: 'GET',
            mode: 'no-cors',
            headers: generateHeaders(token),
        });
        return response.json();
    }
    // else
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/list`, {
        method: 'GET',
        mode: 'no-cors',
        headers: generateHeaders(token),
    });
    return response.json();
}

export async function updateMember(token: string, memberID: number, memberData: object) {
    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/member/${memberID}`, {
        method: 'PATCH',
        mode: 'no-cors',
        headers: generateHeaders(token),
        body: JSON.stringify(memberData),
    });
    return response.json();
}