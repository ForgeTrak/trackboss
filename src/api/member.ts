import { Request, Response, Router } from 'express';

const member = Router();

member.post('/new', (req: Request, res: Response) => {
    res.status(501).send();
});

member.get('/list', (req: Request, res: Response) => {
    res.status(501).send();
});

member.get('/:memberID', (req: Request, res: Response) => {
    res.status(501).send();
});

member.patch('/:memberID', (req: Request, res: Response) => {
    res.status(501).send();
});

member.delete('/:memberID', (req: Request, res: Response) => {
    res.status(501).send();
});

export default member;
