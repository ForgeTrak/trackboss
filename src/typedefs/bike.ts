import { ErrorResponse } from './errorResponse';

export type Bike = {
    bikeId: number,
    year: string,
    make: string,
    model: string,
    membershipAdmin: string
}

export type DeletedBike = {
    bikeId: number
}

export type PostNewBikeRequest = {
    year?: string,
    make?: string,
    model?: string,
    membershipId: number
}

export type PostNewBikeResponse = Bike | ErrorResponse

export type GetBikeRequest = Record<string, never>

export type GetBikeResponse = Bike | ErrorResponse

export type PatchBikeRequest = {
    year?: string,
    make?: string,
    model?: string,
    membershipId?: number
}

export type PatchBikeResponse = Bike | ErrorResponse

export type DeleteBikeRequest = Record<string, never>

export type DeleteBikeResponse = DeletedBike | ErrorResponse

export type GetBikeListRequest = Record<string, never>

export type GetBikeListResponse = Bike[] | ErrorResponse
