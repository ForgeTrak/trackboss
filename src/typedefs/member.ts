import { BoardMember } from './boardMember';
import { ErrorResponse } from './errorResponse';

export type Member = {
    memberId: number,
    membershipId: number,
    membershipAdmin: string,
    uuid: string,
    active: boolean,
    memberTypeId: number,
    memberType: string,
    membershipType: string,
    firstName: string,
    lastName: string,
    phoneNumber: string,
    occupation: string,
    email: string,
    birthdate: string,
    dateJoined: string,
    address: string,
    city: string,
    state: string,
    zip: string,
    lastModifiedDate: string,
    lastModifiedBy: string,
    boardMemberData? : BoardMember // optional field to be used only by the frontend to store data
}

export type PostNewMemberRequest = {
    membershipId?: number,
    uuid?: string,
    memberTypeId: number,
    firstName?: string,
    lastName?: string,
    phoneNumber?: string,
    occupation?: string,
    email?: string,
    birthdate?: string,
    dateJoined?: string,
    modifiedBy: number
}

export type PostNewMemberResponse = Member | ErrorResponse;

export type GetMemberRequest = Record<string, never>;

export type GetMemberResponse = Member | ErrorResponse;

export type PatchMemberRequest = {
    membershipId?: number,
    uuid?: string,
    active?: boolean,
    memberTypeId?: number,
    firstName?: string,
    lastName?: string,
    phoneNumber?: string,
    occupation?: string,
    email?: string,
    birthdate?: string,
    dateJoined?: string,
    modifiedBy: number
}

export type PatchMemberResponse = Member | ErrorResponse;

export type GetMemberListRequest = Record<string, never>

export type GetMemberListResponse = Member[] | ErrorResponse;

export type GetMemberListFilters = {
    type?: string,
    membershipId?: number
}
