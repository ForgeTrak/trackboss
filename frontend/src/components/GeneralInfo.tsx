import React, { useEffect, useState, useContext } from 'react';
import {
    Text,
    Heading,
    VStack,
    HStack,
    Divider,
    SimpleGrid,
    Button,
    Input,
} from '@chakra-ui/react';
import { Member, PatchMemberRequest } from '../../../src/typedefs/member';
import { updateMember } from '../controller/member';
import { UserContext } from '../contexts/UserContext';

interface cardProps {
    user: Member,
}

export default function GeneralInfo(props: cardProps) {
    const { state, update } = useContext(UserContext);
    const [memberInfo, setMemberInfo] = useState<Member>();
    const [editingMemberInfo, setEditingMemberInfo] = useState<boolean>(false);
    const [editedName, setEditedName] = useState<string>('');
    const [editedEmail, setEditedEmail] = useState<string>('');
    const [editedPhone, setEditedPhone] = useState<string>('');
    const [error, setError] = useState<string>('');

    const handleEditedNameChange = (event: { target: { value: any; }; }) => setEditedName(event.target.value);
    const handleEditedEmailChange = (event: { target: { value: any; }; }) => setEditedEmail(event.target.value);
    const handleEditedPhoneChange = (event: { target: { value: any; }; }) => setEditedPhone(event.target.value);

    async function handlePatchMemberContactInfo(
        memberPatchInfo: Member,
        name: string | undefined,
        email: string | undefined,
        phone: string | undefined,
    ) {
        const nameArray = name?.split(' ');
        if (nameArray && state.user) {
            const first = nameArray[0];
            const last = nameArray[1];
            const patch: PatchMemberRequest = {
                membershipId: memberPatchInfo.membershipId,
                uuid: memberPatchInfo.uuid,
                memberTypeId: memberPatchInfo.memberTypeId,
                firstName: first || memberPatchInfo.firstName,
                lastName: last || memberPatchInfo.lastName,
                phoneNumber: phone || memberPatchInfo.phoneNumber,
                occupation: memberPatchInfo.occupation,
                email: email || memberPatchInfo.email,
                birthdate: memberPatchInfo.birthdate,
                dateJoined: memberPatchInfo.dateJoined,
                modifiedBy: memberPatchInfo.memberId,
            };

            const res = await updateMember(state.token, state.user.memberId, patch);
            if ('reason' in res) {
                setError(res.reason);
            } else {
                update({ loggedIn: true, token: state.token, user: res, storedUser: undefined });
                setMemberInfo(res);
            }
        }
        return undefined;
    }

    useEffect(() => {
        async function setMemberData() {
            setMemberInfo(props.user);
        }
        setMemberData();
    }, [props.user]);

    return (
        <VStack mt={25}>
            <HStack>
                { error !== '' && ({ error })}
                <Heading>General Information</Heading>
                {
                    props.user.memberType === 'Admin' && (
                        <Button
                            mr={400}
                            textDecoration="underline"
                            color="orange"
                            variant="ghost"
                            size="lg"
                            onClick={
                                () => {
                                    if (editingMemberInfo) {
                                        setEditingMemberInfo(false);
                                    } else {
                                        setEditingMemberInfo(true);
                                    }
                                }
                            }
                        >
                            Edit
                        </Button>
                    )
                }
            </HStack>
            <Divider />
            {
                memberInfo && (
                    <SimpleGrid pt={10} columns={2}>
                        <VStack spacing={2} align="left">
                            <Text fontSize="3xl" fontWeight="bold">Name:</Text>
                            <Text fontSize="3xl" fontWeight="bold">Email:</Text>
                            <Text fontSize="3xl" fontWeight="bold">Phone:</Text>
                            <Text fontSize="3xl" fontWeight="bold">Joined:</Text>
                        </VStack>
                        {
                            editingMemberInfo ? (
                                <VStack mt={1.5} ml="-50px" align="left" spacing={3}>
                                    <Input
                                        placeholder={`${memberInfo.firstName} ${memberInfo.lastName}`}
                                        value={editedName}
                                        onChange={handleEditedNameChange}
                                        size="md"
                                    />
                                    <Input
                                        placeholder={memberInfo.email}
                                        value={editedEmail}
                                        onChange={handleEditedEmailChange}
                                        size="md"
                                    />
                                    <Input
                                        placeholder={memberInfo.phoneNumber}
                                        value={editedPhone}
                                        onChange={handleEditedPhoneChange}
                                        size="md"
                                    />
                                    <Text fontSize="3xl">{memberInfo.dateJoined}</Text>
                                    <Button
                                        ml={10}
                                        variant="outline"
                                        size="md"
                                        color="green"
                                        onClick={
                                            async () => {
                                                // eslint-disable-next-line max-len
                                                await handlePatchMemberContactInfo(memberInfo, editedName, editedEmail, editedPhone);
                                                setEditingMemberInfo(false);
                                            }
                                        }
                                    >
                                        Save
                                    </Button>
                                </VStack>
                            ) : (
                                <VStack ml="-70px" align="left">
                                    <Text fontSize="3xl">
                                        {`${memberInfo.firstName} ${memberInfo.lastName}`}
                                    </Text>
                                    <Text fontSize="3xl">{memberInfo.email}</Text>
                                    <Text fontSize="3xl">{memberInfo.phoneNumber}</Text>
                                    <Text fontSize="3xl">{memberInfo.dateJoined}</Text>
                                </VStack>
                            )
                        }
                    </SimpleGrid>
                )
            }
        </VStack>
    );
}
