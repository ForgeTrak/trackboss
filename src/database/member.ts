import _ from 'lodash';
import { OkPacket, RowDataPacket } from 'mysql2';

import logger from '../logger';
import { getPool } from './pool';
import { GetMemberListFilters, Member, PatchMemberRequest, PostNewMemberRequest } from '../typedefs/member';
import { createCognitoUser } from '../util/cognito';

// Map the API values for the member types to the DB values
export const MEMBER_TYPE_MAP = new Map([
    ['admin', 'Admin'],
    ['membershipAdmin', 'Membership Admin'],
    ['member', 'Member'],
    ['paidLaborer', 'Paid Laborer'],
]);

export const GET_MEMBER_LIST_SQL = 'SELECT * FROM v_member';
export const GET_MEMBER_SQL = `${GET_MEMBER_LIST_SQL} WHERE member_id = ?`;
export const GET_MEMBER_UUID_SQL = `${GET_MEMBER_LIST_SQL} WHERE uuid = ?`;
export const INSERT_MEMBER_SQL = 'INSERT INTO member (membership_id, uuid, member_type_id, first_name, last_name, ' +
    'phone_number, occupation, email, birthdate, date_joined, last_modified_date, last_modified_by, active) ' +
    'VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, CURDATE(), ?, 1)';
export const PATCH_MEMBER_SQL = 'CALL sp_patch_member(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)';
export const GET_VALID_ACTORS_SQL = 'select member_id from pradb.member m where member_id=? or (m.member_type_id=2 ' +
    'and m.member_id=(select ms.membership_admin_id from pradb.member m left join pradb.membership ms on ' +
    'm.membership_id=ms.membership_id where member_id=?)) or m.member_type_id=1';

export async function insertMember(req: PostNewMemberRequest): Promise<number> {
    if (req.email) {
        logger.info(`Creating new user for email ${req.email} on membership ${req.membershipId}`);
        try {
            const isMembershipAdmin = (req.memberTypeId === 8);
            const uuid = await createCognitoUser(req.email, isMembershipAdmin);
            req.uuid = uuid;
            logger.info(`Successfully created ${req.email} in cognito as ${uuid}`);
        } catch (error: any) {
            logger.error(`Failure creating ${req.email} in cognito.  Continuing on trackboss database side`);
            logger.error(error);
        }
    }
    const values = [
        req.membershipId,
        req.uuid,
        req.memberTypeId,
        req.firstName,
        req.lastName,
        req.phoneNumber,
        req.occupation,
        req.email,
        req.birthdate,
        req.dateJoined,
        req.modifiedBy,
    ];

    let result;
    try {
        [result] = await getPool().query<OkPacket>(INSERT_MEMBER_SQL, values);
    } catch (e: any) {
        if ('errno' in e) {
            switch (e.errno) {
                case 1048: // non-null violation, missing a non-nullable column
                case 1452: // FK violation - referenced is missing
                    logger.error(`User error inserting member in DB: ${e}`);
                    throw new Error('user input error');
                default:
                    logger.error(`DB error inserting member: ${e}`);
                    throw new Error('internal server error');
            }
        } else {
            // this should not happen - errors from query should always have 'errno' field
            throw e;
        }
    }

    return result.insertId;
}

export async function getMemberList(filters: GetMemberListFilters): Promise<Member[]> {
    let sql;
    let values: any[] = [];
    if (!_.isEmpty(filters)) {
        let dynamicSql = ' WHERE ';
        let counter = 0;
        if (typeof filters.type !== 'undefined') {
            dynamicSql += 'member_type = ? AND ';
            // if type is not in the map, it won't hurt to throw it in
            // anyway (and this makes testing easier)
            values[counter++] = MEMBER_TYPE_MAP.get(filters.type) || filters.type;
        }
        if (typeof filters.membershipId !== 'undefined') {
            dynamicSql += 'membership_id = ? AND ';
            values[counter++] = filters.membershipId;
        }
        sql = GET_MEMBER_LIST_SQL + dynamicSql.slice(0, -4); // slice the trailing AND
    } else {
        sql = GET_MEMBER_LIST_SQL;
        values = [];
    }
    let results;
    try {
        [results] = await getPool().query<RowDataPacket[]>(sql, values);
    } catch (e) {
        logger.error(`DB error getting member list: ${e}`);
        throw new Error('internal server error');
    }

    return results.map((result) => ({
        memberId: result.member_id,
        membershipId: result.membership_id,
        firstName: result.first_name,
        lastName: result.last_name,
        membershipAdmin: result.membership_admin,
        membershipAdminId: result.membership_admin_id,
        isBoardMember: !!result.board_title_id,
        uuid: result.uuid,
        active: !!result.active[0],
        memberTypeId: result.member_type_id,
        memberType: result.member_type,
        membershipType: result.membership_type,
        phoneNumber: result.phone_number,
        occupation: result.occupation,
        email: result.email,
        birthdate: result.birthdate,
        dateJoined: result.date_joined,
        address: result.address,
        city: result.city,
        state: result.state,
        zip: result.zip,
        lastModifiedDate: result.last_modified_date,
        lastModifiedBy: result.last_modified_by,
    }));
}

export async function getMember(searchParam: string): Promise<Member> {
    const id = Number(searchParam);
    let sql;
    let values;
    let results;
    if (Number.isNaN(id)) {
        // uuid search
        values = [searchParam];
        sql = GET_MEMBER_UUID_SQL;
    } else {
        values = [id];
        sql = GET_MEMBER_SQL;
    }

    try {
        [results] = await getPool().query<RowDataPacket[]>(sql, values);
    } catch (e) {
        logger.error(`DB error getting member: ${e}`);
        throw new Error('internal server error');
    }

    if (_.isEmpty(results)) {
        throw new Error('not found');
    }

    return {
        memberId: results[0].member_id,
        membershipId: results[0].membership_id,
        firstName: results[0].first_name,
        lastName: results[0].last_name,
        membershipAdmin: results[0].membership_admin,
        membershipAdminId: results[0].membership_admin_id,
        isBoardMember: !!results[0].board_title_id,
        uuid: results[0].uuid,
        active: (results[0].active > 0),
        memberTypeId: results[0].member_type_id,
        memberType: results[0].member_type,
        membershipType: results[0].membership_type,
        phoneNumber: results[0].phone_number,
        occupation: results[0].occupation,
        email: results[0].email,
        birthdate: results[0].birthdate,
        dateJoined: results[0].date_joined,
        address: results[0].address,
        city: results[0].city,
        state: results[0].state,
        zip: results[0].zip,
        lastModifiedDate: results[0].last_modified_date,
        lastModifiedBy: results[0].last_modified_by,
    };
}

export async function getMemberByPhone(phone: string): Promise<Member> {
    let results;
    try {
        const sql = 'SELECT * FROM pradb.v_member where phone_number = ?';
        [results] = await getPool().query<RowDataPacket[]>(sql, [phone]);
    } catch (e) {
        logger.error(e);
        throw e;
    }
    return {
        memberId: results[0].member_id,
        membershipId: results[0].membership_id,
        firstName: results[0].first_name,
        lastName: results[0].last_name,
        membershipAdmin: results[0].membership_admin,
        membershipAdminId: results[0].membership_admin_id,
        isBoardMember: !!results[0].board_title_id,
        uuid: results[0].uuid,
        active: true,
        memberTypeId: results[0].member_type_id,
        memberType: results[0].member_type,
        membershipType: results[0].membership_type,
        phoneNumber: results[0].phone_number,
        occupation: results[0].occupation,
        email: results[0].email,
        birthdate: results[0].birthdate,
        dateJoined: results[0].date_joined,
        address: results[0].address,
        city: results[0].city,
        state: results[0].state,
        zip: results[0].zip,
        lastModifiedDate: results[0].last_modified_date,
        lastModifiedBy: results[0].last_modified_by,
    };
}

export async function getMemberByEmail(email: string): Promise<any> {
    let results;
    try {
        const sql = `SELECT * FROM pradb.v_member where email like '%${email}%'`;
        [results] = await getPool().query<RowDataPacket[]>(sql, [email]);
    } catch (e) {
        logger.error(e);
        throw e;
    }
    let memberWithEmail = {};
    if (results?.length > 0) {
        memberWithEmail = {
            memberId: results[0].member_id,
            membershipId: results[0].membership_id,
            firstName: results[0].first_name,
            lastName: results[0].last_name,
            membershipAdmin: results[0].membership_admin,
            membershipAdminId: results[0].membership_admin_id,
            uuid: results[0].uuid,
            active: true,
            memberTypeId: results[0].member_type_id,
            memberType: results[0].member_type,
            membershipType: results[0].membership_type,
            phoneNumber: results[0].phone_number,
            occupation: results[0].occupation,
            email: results[0].email,
            birthdate: results[0].birthdate,
            dateJoined: results[0].date_joined,
            address: results[0].address,
            city: results[0].city,
            state: results[0].state,
            zip: results[0].zip,
            lastModifiedDate: results[0].last_modified_date,
            lastModifiedBy: results[0].last_modified_by,
        };
    }
    return memberWithEmail;
}

export async function patchMember(id: string, req: PatchMemberRequest): Promise<void> {
    if (_.isEmpty(req)) { // empty request
        throw new Error('user input error');
    }
    const values = [
        id,
        req.membershipId,
        req.uuid,
        req.active,
        req.memberTypeId,
        req.firstName,
        req.lastName,
        req.phoneNumber,
        req.occupation,
        req.email,
        req.birthdate,
        req.dateJoined,
        req.modifiedBy,
    ];
    let result;
    try {
        [result] = await getPool().query<OkPacket>(PATCH_MEMBER_SQL, values);
    } catch (e: any) {
        if ('errno' in e) {
            switch (e.errno) {
                case 1451: // FK violation - referenced somewhere else
                case 1452: // FK violation - referenced is missing
                    logger.error(`User error patching member in DB: ${e}`);
                    throw new Error('user input error');
                default:
                    logger.error(`DB error patching member: ${e}`);
                    throw new Error('internal server error');
            }
        } else {
            // this should not happen - errors from query should always have 'errno' field
            throw e;
        }
    }

    if (result.affectedRows < 1) {
        throw new Error('not found');
    }
}

export async function getValidActors(member: number): Promise<number[]> {
    const values = [member, member];
    let results;
    try {
        [results] = await getPool().query<RowDataPacket[]>(GET_VALID_ACTORS_SQL, values);
    } catch (e) {
        logger.error(`DB error getting valid actors: ${e}`);
        throw new Error('internal server error');
    }
    if (_.isEmpty(results)) {
        throw new Error('not found');
    }
    return _.map(results, (result) => result.member_id);
}

export async function deleteFamilyMember(memberId: number): Promise<number> {
    // remove the member but only if they are a sub member. Magic numberism here, fix it later.
    const sql =
        'delete from member where member_id = ? and member_type_id = 9';
    const values = [memberId];

    let result;
    try {
        [result] = await getPool().query<OkPacket>(sql, values);
    } catch (e) {
        logger.error(`DB error deleting event: ${e}`);
        throw new Error('internal server error');
    }
    return result.affectedRows;
}
