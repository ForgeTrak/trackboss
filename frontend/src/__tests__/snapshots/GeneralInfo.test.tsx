import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import GeneralInfo from '../../components/GeneralInfo';
import { Member } from '../../../../src/typedefs/member';

const member: Member = {
    memberId: 1,
    membershipAdmin: 'true',
    active: true,
    memberType: 'Admin',
    firstName: 'test',
    lastName: 'member',
    phoneNumber: '1234',
    email: 'user@example.com',
    uuid: '',
    occupation: 'secretary',
    birthdate: '',
    dateJoined: '',
    address: '',
    city: '',
    state: '',
    zip: '',
    lastModifiedDate: '',
    lastModifiedBy: '',
};

const nonAdminMember: Member = {
    memberId: 1,
    membershipAdmin: 'true',
    active: true,
    memberType: 'user',
    firstName: 'test',
    lastName: 'member',
    phoneNumber: '1234',
    email: 'user@example.com',
    uuid: '',
    occupation: 'secretary',
    birthdate: '',
    dateJoined: '',
    address: '',
    city: '',
    state: '',
    zip: '',
    lastModifiedDate: '',
    lastModifiedBy: '',
};

describe('general info', () => {
    it('renders all props correctly as admin', () => {
        const modal = render(
            <GeneralInfo user={member} />,
        );
        expect(modal).toMatchSnapshot();
    });

    it('renders all props correctly as not admin', () => {
        const modal = render(
            <GeneralInfo user={nonAdminMember} />,
        );
        expect(modal).toMatchSnapshot();
    });
});
