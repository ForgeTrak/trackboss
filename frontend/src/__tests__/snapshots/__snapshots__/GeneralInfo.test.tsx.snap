// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`general info renders all props correctly as admin 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-rtxd0z"
      >
        <div
          class="chakra-stack css-84zodg"
        >
          <h2
            class="chakra-heading css-0"
          >
            General Information
          </h2>
          <button
            class="chakra-button css-aab1vq"
            type="button"
          >
            Edit
          </button>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-1upb9tn"
        />
        <div
          class="css-1nfywic"
        >
          <div
            class="chakra-stack css-3w7d23"
          >
            <p
              class="chakra-text css-lnxk8v"
            >
              Name:
            </p>
            <p
              class="chakra-text css-lnxk8v"
            >
              Email:
            </p>
            <p
              class="chakra-text css-lnxk8v"
            >
              Phone:
            </p>
            <p
              class="chakra-text css-lnxk8v"
            >
              Joined:
            </p>
          </div>
          <div
            class="chakra-stack css-14nvnon"
          >
            <p
              class="chakra-text css-1tapsqs"
            >
              test member
            </p>
            <p
              class="chakra-text css-1tapsqs"
            >
              user@example.com
            </p>
            <p
              class="chakra-text css-1tapsqs"
            >
              1234
            </p>
            <p
              class="chakra-text css-1tapsqs"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-rtxd0z"
    >
      <div
        class="chakra-stack css-84zodg"
      >
        <h2
          class="chakra-heading css-0"
        >
          General Information
        </h2>
        <button
          class="chakra-button css-aab1vq"
          type="button"
        >
          Edit
        </button>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="css-1nfywic"
      >
        <div
          class="chakra-stack css-3w7d23"
        >
          <p
            class="chakra-text css-lnxk8v"
          >
            Name:
          </p>
          <p
            class="chakra-text css-lnxk8v"
          >
            Email:
          </p>
          <p
            class="chakra-text css-lnxk8v"
          >
            Phone:
          </p>
          <p
            class="chakra-text css-lnxk8v"
          >
            Joined:
          </p>
        </div>
        <div
          class="chakra-stack css-14nvnon"
        >
          <p
            class="chakra-text css-1tapsqs"
          >
            test member
          </p>
          <p
            class="chakra-text css-1tapsqs"
          >
            user@example.com
          </p>
          <p
            class="chakra-text css-1tapsqs"
          >
            1234
          </p>
          <p
            class="chakra-text css-1tapsqs"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`general info renders all props correctly as not admin 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-rtxd0z"
      >
        <div
          class="chakra-stack css-84zodg"
        >
          <h2
            class="chakra-heading css-0"
          >
            General Information
          </h2>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-1upb9tn"
        />
        <div
          class="css-1nfywic"
        >
          <div
            class="chakra-stack css-3w7d23"
          >
            <p
              class="chakra-text css-lnxk8v"
            >
              Name:
            </p>
            <p
              class="chakra-text css-lnxk8v"
            >
              Email:
            </p>
            <p
              class="chakra-text css-lnxk8v"
            >
              Phone:
            </p>
            <p
              class="chakra-text css-lnxk8v"
            >
              Joined:
            </p>
          </div>
          <div
            class="chakra-stack css-14nvnon"
          >
            <p
              class="chakra-text css-1tapsqs"
            >
              test member
            </p>
            <p
              class="chakra-text css-1tapsqs"
            >
              user@example.com
            </p>
            <p
              class="chakra-text css-1tapsqs"
            >
              1234
            </p>
            <p
              class="chakra-text css-1tapsqs"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-rtxd0z"
    >
      <div
        class="chakra-stack css-84zodg"
      >
        <h2
          class="chakra-heading css-0"
        >
          General Information
        </h2>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="css-1nfywic"
      >
        <div
          class="chakra-stack css-3w7d23"
        >
          <p
            class="chakra-text css-lnxk8v"
          >
            Name:
          </p>
          <p
            class="chakra-text css-lnxk8v"
          >
            Email:
          </p>
          <p
            class="chakra-text css-lnxk8v"
          >
            Phone:
          </p>
          <p
            class="chakra-text css-lnxk8v"
          >
            Joined:
          </p>
        </div>
        <div
          class="chakra-stack css-14nvnon"
        >
          <p
            class="chakra-text css-1tapsqs"
          >
            test member
          </p>
          <p
            class="chakra-text css-1tapsqs"
          >
            user@example.com
          </p>
          <p
            class="chakra-text css-1tapsqs"
          >
            1234
          </p>
          <p
            class="chakra-text css-1tapsqs"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
